plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTableProperties.java

```diff
@@ -73,6 +76,18 @@ public DeltaLakeTableProperties()
                         "Enables storing change data feed entries",
                         null,
                         false))
+                .add(enumProperty(
+                        COLUMN_MAPPING_MODE_PROPERTY,
+                        "Column mapping mode",
+                        ColumnMappingMode.class,
```

Comment:
nit: no actual bug - There can't be created tables with `UNKNOWN` column mapping mode
```
trino> create table delta.tiny.table3 (c integer) with (column_mapping_mode = 'fdasf');
Query 20230607_103926_00005_zb4u4 failed: Unable to set catalog 'delta' table property 'column_mapping_mode' to ['fdasf']: Invalid value [fdasf]. Valid values: [ID, NAME, NONE, UNKNOWN]
io.trino.spi.TrinoException: Unable to set catalog 'delta' table property 'column_mapping_mode' to ['fdasf']: Invalid value [fdasf]. Valid values: [ID, NAME, NONE, UNKNOWN]
```

`UNKNOWN` is not actually valid.

We'd probably need a slightly different version of `io.trino.spi.session.PropertyMetadata#enumProperty(java.lang.String, java.lang.String, java.lang.Class<T>, T, java.util.function.Consumer<T>, boolean)` which filters out from the values that can be chosen.

Reply:
instead of changing PropertyMetadata, let's use a different (dedicated) enum for the table property
(OR use varchar property and decode manually.)